File: weather-app-frontend/src/servises/WeatherService.js
// src/services/WeatherService.js

import { fetchCurrentWeather, fetchForecast } from '../api/api';

class WeatherService {
    // Method to get current weather
    async getCurrentWeather(city, country) {
        try {
            const data = await fetchCurrentWeather(city, country);
            return data;
        } catch (error) {
            throw new Error('Failed to fetch current weather: ' + error.message);
        }
    }

    // Method to get 16-day weather forecast
    async getForecast(city, country) {
        try {
            const data = await fetchForecast(city, country);
            return data;
        } catch (error) {
            throw new Error('Failed to fetch weather forecast: ' + error.message);
        }
    }
}

export default new WeatherService();

File: src/main/java/com/app/weather/services/WeatherService.java
package com.app.weather.services;

import com.app.weather.model.pojo.ForecastData;
import com.app.weather.model.pojo.WeatherData;
import org.springframework.stereotype.Service;


public interface WeatherService {
    WeatherData getCurrentWeather(String city, String country);
    ForecastData[] getForecast(String city, String country);
}
